src/raster/rasterband.rs

```diff
@@ -713,6 +713,101 @@ impl<'a> RasterBand<'a> {
+        let mut pb_success = 1;
+        let no_data = unsafe {
+            gdal_sys::GDALGetRasterNoDataValueAsUInt64(self.c_rasterband, &mut pb_success)
+        };
+        if pb_success == 1 {
+            return Some(no_data);
+        }
+        None
+    }
+
+    /// Sets the no-data value for a `UInt64` band.
+    ///
+    /// This method should ONLY be called on bands whose data type is `UInt64`.
+    ///
+    /// If `no_data` is `None`, any existing no-data value is deleted.
+    ///
+    /// # Notes
+    /// See also:
+    /// [`GDALSetRasterNoDataValueAsUInt64`](https://gdal.org/api/raster_c_api.html#_CPPv432GDALSetRasterNoDataValueAsUInt6415GDALRasterBandH8uint64_t),
+    /// [`GDALDeleteRasterNoDataValue`](https://gdal.org/api/raster_c_api.html#_CPPv427GDALDeleteRasterNoDataValue15GDALRasterBandH)
+    #[cfg(all(major_ge_3, minor_ge_5))]
+    pub fn set_no_data_value_u64(&mut self, no_data: Option<u64>) -> Result<()> {
+        let rv = if let Some(no_data) = no_data {
+            unsafe { gdal_sys::GDALSetRasterNoDataValueAsUInt64(self.c_rasterband, no_data) }
+        } else {
+            unsafe { gdal_sys::GDALDeleteRasterNoDataValue(self.c_rasterband) }
+        };
+
+        if rv != CPLErr::CE_None {
+            Err(_last_cpl_err(rv))
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Fetch the no-data value for this band.
+    ///
+    /// This method should ONLY be called on bands whose data type is `Int64`.
+    ///
+    /// The no data value returned is 'raw', meaning that it has no offset and scale applied.
+    ///
+    /// # Returns
+    /// No-data value as `Some(i64)` if no-data value exists, `None` otherwise.
+    ///
+    /// # Notes
+    /// See also: [`GDALGetRasterNoDataValueAsInt64`](https://gdal.org/api/gdalrasterband_cpp.html#_CPPv4N14GDALRasterBand21GetNoDataValueAsInt64EPi)
+    #[cfg(all(major_ge_3, minor_ge_5))]
+    pub fn no_data_value_i64(&self) -> Option<i64> {
+        let mut pb_success = 1;
+        let no_data = unsafe {
+            gdal_sys::GDALGetRasterNoDataValueAsInt64(self.c_rasterband, &mut pb_success)
+        };
+        if pb_success == 1 {
+            return Some(no_data);
+        }
+        None
+    }
+
+    /// Sets the no-data value for a `Int64` band.
+    ///
+    /// This method should ONLY be called on bands whose data type is `Int64`.
+    ///
+    /// If `no_data` is `None`, any existing no-data value is deleted.
+    ///
+    /// # Notes
+    /// See also:
+    /// [`GDALSetRasterNoDataValueAsInt64`](https://gdal.org/api/raster_c_api.html#_CPPv431GDALSetRasterNoDataValueAsInt6415GDALRasterBandH7int64_t),
+    /// [`GDALDeleteRasterNoDataValue`](https://gdal.org/api/raster_c_api.html#_CPPv427GDALDeleteRasterNoDataValue15GDALRasterBandH)
+    #[cfg(all(major_ge_3, minor_ge_5))]
+    pub fn set_no_data_value_i64(&mut self, no_data: Option<i64>) -> Result<()> {
+        let rv = if let Some(no_data) = no_data {
```

Comment:
Well, isn't [this](https://github.com/georust/gdal/pull/308#discussion_r964887637) awkward...

Taking an `Option` seemed like a good idea at the time...

Reply:
I still think there's value in keeping them consistent...

The alternative is to add [GDALDeleteRasterNoDataValue](https://gdal.org/api/raster_c_api.html#_CPPv427GDALDeleteRasterNoDataValue15GDALRasterBandH)

Reply:
And now I think we should go for that (but in a separate PR).